### PR TITLE
tmpfiles: remove children before their parents plus other fixlets

### LINKED ISSUE #9508: systemd-tmpfiles: create and remove in wrong order
**systemd version the issue has been seen with**
237

**Used distribution**
openSUSE Tumbleweed

**Expected behaviour you didn't see**
The manual page for systemd-tmpfiles states:
   For example, during boot the following command line is executed to
  ensure that all temporary and volatile directories are removed and
  created according to the configuration file:

  systemd-tmpfiles --remove --create

The expected and most logical behavior is, that at first the old stuff is removed and afterwards a clean structure is created.

**Unexpected behaviour you saw**
At first, the new files are created and only afterwards the other ones are deleted. This requires pretty complex regex expressions for the cleanup to not by accident delete the new stuff.

**Steps to reproduce the problem**
Create a simple config with:
d /var/tmp/test
d /var/tmp/test/important
R /var/tmp/test

Create a directory /var/tmp/test with several files in it. 

Run "systemd-tmpfiles --remove --create test.conf". Afterwards, /var/tmp/test and /var/tmp/test/important don't exist. Expected behavior is, that all other files in /var/tmp/test are removed (including important directory), and a new /var/tmp/test/important directory is created.

### PR DESCRIPTION
**Pull request by poettering**
Fixes #9508

### CODE REVIEW

**Comment by evverx** [comment]
This pull request **introduces 1 alert** when merging e8e41017b63f6bd3d3c70e15aa3a2b4f30a59ac9 into bb01aab6ef4dc500cdbc72ccb66c5ef12b8aedfc - [view on LGTM.com](https://lgtm.com/projects/g/systemd/systemd/rev/pr-156d77fabaa9584e54149e3c7008aabd5d3541d9)

**new alerts:**

* 1 for FIXME comment

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by evverx** [comment]
This pull request **introduces 1 alert** when merging 0cfabb8ef7213505b174c8c71c946bdb3e63d062 into bb01aab6ef4dc500cdbc72ccb66c5ef12b8aedfc - [view on LGTM.com](https://lgtm.com/projects/g/systemd/systemd/rev/pr-15fd5a3bde1c72c7db75bf083f0014c2c46172fc)

**new alerts:**

* 1 for FIXME comment

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by evverx** [comment]
This pull request **introduces 1 alert** when merging ad19c57898b96ecae39990e849a865db5b94fe4d into 8d2411f69348a081419c2fb7cddbe1970cf3eac5 - [view on LGTM.com](https://lgtm.com/projects/g/systemd/systemd/rev/pr-58524cf08e9b55c2b41ade7460959e92914f132b)

**new alerts:**

* 1 for FIXME comment

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/tmpfiles/tmpfiles.c [hunk 1]
```diff
@@ -2258,8 +2258,12 @@ static int process_item(Item *i, OperationMask operation) {
 
         i->done |= operation;
 
-        if (chase_symlinks(i->path, NULL, CHASE_NO_AUTOFS, NULL) == -EREMOTE)
+        r = chase_symlinks(i->path, NULL, CHASE_NO_AUTOFS, NULL);
+        if (r == -EREMOTE) {
+                log_debug_errno(r, "Item '%s' is behind autofs, skipping.", i->path);
```
Comment (keszybz) [inline src/tmpfiles/tmpfiles.c:2271]: behind → below?

File: man/tmpfiles.d.xml [hunk 2]
```diff
@@ -67,28 +67,20 @@
...
+    <filename>/run/tmpfiles.d</filename> override files with the same name in
+    <filename>/usr/lib/tmpfiles.d</filename>. Packages should install their configuration files in
+    <filename>/usr/lib/tmpfiles.d</filename>. Files in <filename>/etc/tmpfiles.d</filename> are reserved for the local
+    administrator, who may use this logic to override the configuration files installed by vendor packages. All
+    configuration files are sorted by their filename in lexicographic order, regardless of which of the directories
+    they reside in. If multiple files specify the same path, the entry in the file with the lexicographically earliest
+    name will be applied.  All other conflicting entries will be logged as errors. When two lines are prefix path and
+    suffix path of each other, then the prefix line is always created first, the suffix later (and if removal applies
+    to the line, the order is reversed: the suffix is removed first, the prefix later). Lines that take globs are
+    applied after those accepting no globs. If multiple operations shall be applied on the same file, (such as ACL,
```
Comment (keszybz) [inline man/tmpfiles.d.xml:81]: No comma before the opening parenthesis.

File: man/tmpfiles.d.xml [hunk 3]
```diff
@@ -67,28 +67,20 @@
...
+    xattr, file attribute adjustments), these are always done in the same fixed order. Otherwise, the files/directories
```
Comment (keszybz) [inline man/tmpfiles.d.xml:82]: "Otherwise" → "Except for those cases"

**Pull request merged**